src/phase0/state_transition/context.rs

```diff
@@ -69,7 +69,7 @@ impl Context {
             max_seed_lookahead: preset.max_seed_lookahead,
             min_epochs_to_inactivity_penalty: preset.min_epochs_to_inactivity_penalty,
             epochs_per_eth1_voting_period: preset.epochs_per_eth1_voting_period,
-            slots_per_historical_root: preset.slots_per_historical_root,
+            slots_per_historical_root: preset.slots_per_historical_root as Slot,
```

Comment:
If I don't cast this as a `Slot`, then I'd have to mess with the `const`s in `presets/mainnet.rs` (e.g., change `SLOTS_PER_HISTORICAL_ROOT` from `usize` to `u64`). Just calling out if there's a reason this isn't ideal, but the former was obviously more straightforward than the latter.

Reply:
yep nice catch....

so ideally we would not need to be so precise w/ the type of the const generic and that is what i was getting at here: https://github.com/ralexstokes/ethereum_consensus/issues/1

but until we can unify the numeric types somehow (if ever), we will need a cast....

so -- right idea but i would suggest we move this "up" one struct to the `Preset`

so there is the constant `pub const ...: usize = ...;` and then the field in the `Preset` is also a `Slot` and you will need to do this `as Slot` cast there.

i think there are a few places where we can tighten the typing in `Context` / `Preset` (bc they keep cropping up as we implement the rest of the spec) so if you'd like to just raise an issue and tag me i can make a pass across the entire struct -- it seems like something i should have done the first time